### PR TITLE
[SDK] Fix CanaryStrategy. Remove isWaiting()

### PR DESCRIPTION
**Pull request by mbdcos**
Problem: After first continue request, first step should start in Canary. It was not starting any step till all required continues are received.

See commit descriptions:
* Fix Canary Strategy: return phase not interrupted, after first continue, till proceeded step is started.
* Remove isWaiting(). isWaiting() != isInterrupted(), WAITING is a special status, only used to show/print plan/phase/step details.

### CODE REVIEW

**Comment by mgummelt** [comment]
Can you either link a JIRA or provide a description in this PR?

**Comment by mbdcos** [comment]
Problem: After first continue request, first step should start in Canary. It was not starting any step till all required continues are received.

See commit descriptions:
Fix Canary Strategy: return phase not interrupted, after first continue, till proceeded step is started.
Remove isWaiting(). isWaiting() != isInterrupted(), WAITING is a special status, only used to show/print plan/phase/step details.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/CanaryStrategy.java [hunk 1]
```diff
@@ -106,12 +106,24 @@ public void proceed() {
...
     public boolean isInterrupted() {
         Step canaryStep = getNextCanaryStep();
-        if (canaryStep != null) {
+        if (canaryStep != null && getNextProceedStep() == null) {
             return true;
         }
         return strategy.isInterrupted();
     }
 
+    private  Step getNextProceedStep() {
```
Comment (mgummelt) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/CanaryStrategy.java]: remove extra space after `private`

**Comment by mgummelt** [comment]
LGTM

**Pull request merged**